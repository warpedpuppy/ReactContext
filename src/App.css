section {
  display: inline-block;
  border: 5px solid black;
  padding: 10px;
  margin: 10px;
  box-shadow: 0 5px 5px rgba(0,0,0,0.5);
}
button {
  display: block;
  margin: 0 auto;
  padding: 5px 10px;
  border-radius: 5px;
}