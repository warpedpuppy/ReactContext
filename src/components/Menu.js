import React from 'react';
import './Menu.css';

export default function Menu() {
  function buttonHandler(){
    console.log("click")
  }
  return (
    <nav>
      <button onClick={() => buttonHandler()}>button1</button>
      <button onClick={() => buttonHandler()}>button2</button>
      <button onClick={() => buttonHandler()}>button3</button>
    </nav>
  );
}

