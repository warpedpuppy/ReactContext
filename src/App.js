import React from 'react';
import './App.css';
import Menu from './components/Menu.js';
import ColorContext from './ColorContext';

class App extends React.Component {
 state = {
  color: 'orange'
 };
 handleSetColor = (color) => {
   this.setState({ color })
 };

 render() {
        const contextValue = {
         color: this.state.color,
         setColor: this.handleSetColor,
        }
       return (
        <ColorContext.Provider value={contextValue}>
        <div className="App">
          <header className="App-header">
           <Menu />
          </header>
        </div>
        </ColorContext.Provider>
      );
  }
}

export default App;
