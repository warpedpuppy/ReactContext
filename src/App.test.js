import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Menu from './components/Menu.js';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('arbitrary section', () => {
	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<App />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

 

  it('opens any clicked section', () => {
    const wrapper = shallow(<Menu />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  // can I set something up to check whether the above test is just checking if the event happened?
  
})

