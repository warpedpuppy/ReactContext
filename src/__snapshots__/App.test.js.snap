// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arbitrary section opens any clicked section 1`] = `
<nav>
  <button
    onClick={[Function]}
  >
    button1
  </button>
  <button
    onClick={[Function]}
  >
    button2
  </button>
  <button
    onClick={[Function]}
  >
    button3
  </button>
</nav>
`;
